Add build requirements

mpcusack-color opened the pull request:
Adds a `pyproject.toml` containing build time requirements. This is needed because `setup.py` imports from `numpy` which won't be installed in a fresh venv. This made this library (and packages that depend on it- in our case aldy) difficult to package without a custom build process.

With this change doing a `pip install path/to/ssw-py` in a fresh venv works without first installing numpy and cython.

mpcusack-color committed:
Add pyproject.toml specifying build reqs

mpcusack-color committed:
Bump version

aroon-color commented:
@ranikay Sorry for the tag, but we weren't sure who (if anyone) was still actively maintaining this in the Wyss group

mpcusack-color commented:
@grinner :pray:

mpcusack-color committed:
Restrict setuptools version

The most canonical documention of the issue I could find: https://github.com/pypa/setuptools/issues/3693

Also adds a dependency on `wheels` to silence some pip warnings.
https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/#fallback-behaviour
